controllers/util/solr_util.go

```diff
@@ -327,13 +330,42 @@ func GenerateStatefulSet(solrCloud *solr.SolrCloud, solrCloudStatus *solr.SolrCl
 		envVars = append(envVars, customPodOptions.EnvVariables...)
 	}
 
+	// Did the user provide a custom log config?
+	if configMapInfo[LogXmlFile] != "" {
+
+		if configMapInfo[LogXmlMd5Annotation] != "" {
+			if podAnnotations == nil {
+				podAnnotations = make(map[string]string, 1)
+			}
+			podAnnotations[LogXmlMd5Annotation] = configMapInfo[LogXmlMd5Annotation]
+		}
+
+		// cannot use /var/solr as a mountPath, so mount the custom log config in a sub-dir
+		volName := "log4j2-xml"
+		mountPath := fmt.Sprintf("/var/solr/%s-log-config", solrCloud.Name)
+		log4jPropsEnvVarPath := fmt.Sprintf("%s/%s", mountPath, LogXmlFile)
+
+		solrVolumes = append(solrVolumes, corev1.Volume{
```

Comment:
So if the custom configMap contains both a Solr.xml and a log4j.xml, then this will add 2 volumes, using the same configMap, correct?

Is this supported in Kube? I think there might be an issue. But if you've tested it, then maybe it's ok to go.

It would be pretty easy to make sure that the volume is only added once, and it has the correct `items`.

Reply:
This is a good catch! K8s allows it and results in a structure like the following in the STS:
```
      volumes:
      - configMap:
          defaultMode: 420
          items:
          - key: solr.xml
            path: solr.xml
          name: dev-custom-solr-xml
        name: solr-xml
      - configMap:
          defaultMode: 420
          items:
          - key: log4j2.xml
            path: log4j2.xml
          name: dev-custom-solr-xml
        name: log4j2-xml
```
But we certainly should use a single volume with multiple `items` as you suggest ;-) Will fix it up and add a test for both keys being provided.

Reply:
Latest commit addresses this issue.